tools/clone.js

```diff
@@ -22,21 +32,56 @@ async function main() {
     const publishedApps = await request(sourceUrl)
     const appsJson = JSON.parse(publishedApps)
 
+    let targetApps = await request(`${targetUrl}/apps`)
+    let existingAppsAtTarget = JSON.parse(targetApps)
+
+    //TODO: make configurable
+    const cleanTarget = true
+
+    if (cleanTarget) {
+        for (let i = 0; i < existingAppsAtTarget.length; ++i) {
+            const app = existingAppsAtTarget[i]
+            console.log(`Deleting app '${app.name}'`)
+            await request.delete({
+                url: `${targetUrl}/apps/${app.id}`,
+                headers: {
+                    Authorization: 'Bearer ' + authToken,
+                },
+            })
+        }
+        //refresh existing apps
+        targetApps = await request(`${targetUrl}/apps`)
+        existingAppsAtTarget = JSON.parse(targetApps)
+    }
+
     for (let i = 0; i < appsJson.length; ++i) {
         const app = appsJson[i]
 
+        const existingApp = findAppByName(app.name, existingAppsAtTarget)
+
+        if (existingApp !== null) {
+            //app exists, but all versions?
+            console.log(
+                `App "${app.name}" already exists at target/destination, skipping it.`
+            )
+            continue
+        }
+
         if (!fs.existsSync('./apps')) {
             fs.mkdirSync('./apps')
         }
 
-        const dir = './apps/' + app.name
+        const dir = './apps/' + app.name.replace('/', '-')
         if (!fs.existsSync(dir)) {
-            fs.mkdirSync(dir)
+            fs.mkdirSync(dir, '0777', true)
```

Comment:
Isn't world writable a bit much? I would expect `755` to be plenty.

Reply:
`mkdirSync` also takes a function and not 3 params? So this shouldn't do anything other than fallback to default.

Reply:
@varl the default mode is actually 777 so this would only make it more explicit, but of course we can go with 755 instead.

@Birkbjo you are correct, I remembered incorrectly that a third parameter would be recursive creation, but it's passed in a config object. Will change.